src/pagination.rs

```diff
@@ -34,35 +35,32 @@ fn extract_page_path_from(p: &liquid::Object) -> String {
     .into_owned()
 }
 
-pub(crate) fn extract_page_path(p: &liquid::Object, config: &PaginationCfg) -> PathBuf {
-  PathBuf::from(match config.include.as_str() {
-    "all" => extract_page_path_from(&p),
+pub fn extract_page_path(p: &liquid::Object, config: &PaginationConfig) -> PathBuf {
+  PathBuf::from(match config.include {
+    Include::All => extract_page_path_from(&p),
     _ => "".to_owned(),
   })
 }
 
 fn extract_value(a: &liquid::Value, key: &String) -> Option<liquid::Scalar> {
-  a.as_object().map_or(None, |attr| {
-    attr.get(key).map_or(None, |sort_key| {
-      sort_key
-        .as_scalar()
-        .map_or(None, |value| Some(value.clone()))
-    })
-  })
+  a.get(&key.clone().into())
+    .map_or(None, |sort_key| sort_key.as_scalar().cloned())
 }
 
 // sort posts by multiple criteria
-fn sort_posts(posts: &mut Vec<liquid::Value>, config: &PaginationCfg) {
+fn sort_posts(posts: &mut Vec<liquid::Value>, config: &PaginationConfig) {
   posts.sort_by(|a, b| {
-    let keys = config.sort_by.clone();
+    let keys = &config.sort_by;
     let mut cmp = Ordering::Less;
     for k in keys {
       cmp = if let Some(a) = extract_value(a, &k) {
         if let Some(b) = extract_value(b, &k) {
           match config.order {
             SortOrder::Desc => b.partial_cmp(&a).unwrap_or(Ordering::Equal),
             SortOrder::Asc => a.partial_cmp(&b).unwrap_or(Ordering::Equal),
-            SortOrder::None => unreachable!("We should always have a sort order default value"),
+            SortOrder::None => unreachable!(
+              "Sort order should have default value when constructing PaginationConfig"
```

Comment:
This message is incorrect or bad.  A default value is not good enough to protect us from reaching this code path.  We need to explicitly check for the user setting it and disallow it.